jobs/async-upload/tests/test_models.py

```diff
@@ -0,0 +1,273 @@
+        assert config.artifact_id == "test-artifact-id"
+
+    def test_create_model_intent_with_mixed_data_succeeds(self):
+        """Test that create_model intent succeeds with mixed data"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id=""  # Empty string - should be fine
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == "test-model-id"
+        assert config.version_id == "test-version-id"
+        assert config.artifact_id == ""
+
+    def test_create_model_intent_allows_all_empty_ids(self):
+        """Test that create_model intent allows all empty IDs"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="",
+            version_id="",
+            artifact_id=""
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == ""
+        assert config.version_id == ""
+        assert config.artifact_id == ""
+
+    def test_create_version_intent_with_required_ids_succeeds(self):
+        """Test that create_version intent succeeds when model ID and version ID are provided"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_version,
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id="test-artifact-id"  # Can be provided but not required
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_version
+        assert config.id == "test-model-id"
+        assert config.version_id == "test-version-id"
+        assert config.artifact_id == "test-artifact-id"
+
+    def test_create_version_intent_without_artifact_id_succeeds(self):
+        """Test that create_version intent succeeds without artifact ID"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_version,
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id="any-value"  # Not validated for create_version
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_version
+        assert config.id == "test-model-id"
+        assert config.version_id == "test-version-id"
+
+    def test_create_version_intent_missing_model_id_fails(self):
+        """Test that create_version intent fails when model ID is missing"""
+        with pytest.raises(ValidationError) as exc_info:
+            ModelConfig(
+                upload_intent=UploadIntent.create_version,
+                id="",  # Empty string
+                version_id="test-version-id",
+                artifact_id="test-artifact-id"
```

Comment:
... but why these IDs are supplied given the intent is to create the tuple of ModelVersion and ModelArtifact?